dacsspace/validator.py

```diff
@@ -21,6 +21,22 @@ def __init__(self, schema_identifier, schema_filepath):
             self.schema = json.load(json_file)
         self.validator.check_schema(self.schema)
 
+    def format_error(self, error):
+        """Formats validation error messages.
+
+        Args:
+            error (jsonschema.exceptions.ValidationError): a validation error
+            received from jsonschema.
+
+        Returns:
+            message (str): a string representation of the validation error.
+        """
+        if error.validator == "required":
+            return error.message
+        else:
+            schema_path = f"schema[{']['.join(repr(index) for index in error.schema_path)}]"
+            return f"Failed validating {error.validator!r} in {schema_path}: {error.schema}"
```

Comment:
One comment - and something we might want to align - the `!r` here is a conversion field, and specifies that `repr` is called on the specified property rather than `str`. It might be good to look at the JSONschema code to see what the difference is (if any) between the two. Note that above (in line 34) we're simply calling `error.validator` and matching against a string, so that `!r` may be unnecessary.

Reply:
I've tried looking into the code and documentation but can't find enough information on repr to understand the difference. I also tried adjusting the code but keep getting an assertion error back from the validator test

Reply:
Thanks for looking into this @daremiyo. I'd like to propose we make the following change for consistency's sake and so we don't have to remember what `!r` means. Does that work?

```suggestion
            return f"Failed validating {repr(error.validator)} in {schema_path}: {error.schema}"
```